### ISSUE TITLE
Buildasaur hangs on Build Template View when using with Xcode 8

### ISSUE THREAD

**Issue by kdawgwilk** [czechboy0/Buildasaur#294]
# My System Setup

10.11 macOS El Capitan
Xcode 8
Buildasaur v1.0.3
# My Project setup

iOS
CocoaPods
Fastlane
Bundler

I have a few shared schemes as well
4 build configurations (this seems it could likely be the culprit):
- Debug
- Alpha
- Beta
- Release

**Comment by arlo-egg** [comment]
Same here. I can select a scheme from the dropdown but the Build,Test dropdown never gets populated with simulators/devices and the spinner in the top left never stops. 

![screen shot 2016-10-15 at 9 59 52 am](https://cloud.githubusercontent.com/assets/18194305/19411896/e9e6d12e-92be-11e6-8274-6c22f4616fe6.png)

Where could I find logs to help me better understand the problem?

**Comment by MattNewberry** [comment]
+1 Also seeing this as well... In my case, I'm able to see the devices list populated after a period of time, however the "Build,Test" list is never populated.

**Comment by krin-san** [comment]
+1 Do you know any workaround for this issue?

**Comment by krin-san** [comment]
I found something but not sure if it works always. Try to:
* create a new template
* enter template name
* select Scheme

After that just wait some time until device list was filled with devices and "Build,Test" field become non-empty.
I catch this bug when I rushed through the fields filling them with values without waiting :)

**Comment by krin-san** [comment]
Another issue with this case: even if simulators list will be populated, the build template will have an empty name and Syncer could not be started. I'm having issues with syncing simulators list (it works from time to time), with the Next button (it could be disabled sometimes for unknown reason) and even if it's enabled, going to the next page shows empty Template instead of what I created on previous screen.

**Comment by rafalkitta** [comment]
+1 
On macOS Sierra (10.12) and Xcode 8, Buildasaur v1.0.3 is unusable.

**Comment by momamene** [comment]
:+1:

**Comment by Antondomashnev** [comment]
Hi, I've also experienced the same issue, but only with specific project. As a workaround a can suggest to create a template manually. 
Builda stores all configuration files under the folder `~/Library/Application Support/Buildasaur`. There are all json files, with quite straight forward structure. 

Here is an example:
```
{
  "id" : "0D024176-916D-4704-A854-8C21F6C5C30F",
  "project_name" : "CNISDK",
  "name" : "ios-sdk-default",
  "triggers" : [
    "9F1813E1-92ED-4007-8AE9-F2EF33577891",
    "64FA6E0C-F06C-4F08-9582-B599E596953F"
  ],
  "should_analyze" : true,
  "platform_type" : "com.apple.platform.iphoneos",
  "scheme" : "TestAllKits",
  "device_filter" : 3,
  "cleaning_policy" : 0,
  "testing_devices" : [
    "b116b840c4d61aa7defe731d1f01e978"
  ],
  "should_archive" : false,
  "should_test" : true,
  "schedule" : {
    "weeklyScheduleDay" : 0,
    "periodicScheduleInterval" : 0,
    "hourOfIntegration" : 0,
    "minutesAfterHourToIntegrate" : 0,
    "scheduleType" : 2
  }
}
```

The only tricky part is to get the device ID. The way I did it - I copied the device ID from another template, because for another one it worked. However you can use Xcode API to get the list of devices, which is not an easy solution, but work though.

**Comment by krin-san** [comment]
@Antondomashnev, yeah I came to the same workaround. But from my experience, device IDs in Buildasaur template files are not the same as iOS Device IDs you could see in Xcode devices list.

For that I found another workaround: in a signal when Buildasaur obtain JSON with devices list from server, I dump this json to file: https://github.com/buildasaurs/Buildasaur/compare/master...krin-san:feature/bitbucket_lightweight_build?expand=1#diff-542366cda76cec111be456b273308b84

**Comment by Antondomashnev** [comment]
@krin-san hah, nice 👍

**Comment by tomhut** [comment]
@Antondomashnev I also found that by simply GETing `https://your-server.local:20343/api/devices` you will receive a nice JSON payload of devices (including their "_id") from the server. 😄

**Comment by krin-san** [comment]
Buidlasaur hangs on this step because of `xcodebuild -showSettings` hanging bug. Half a day and dozens of verbose Logs helped to locate the traitor 😄
Ref: https://forums.developer.apple.com/thread/50372
